packages/svelte/src/internal/client/dom/blocks/snippet.js

```diff
@@ -38,8 +41,10 @@ export function snippet(node, get_snippet, ...args) {
 			snippet_effect = null;
 		}
 
-		if (snippet) {
+		if (snippet != null) {
 			snippet_effect = branch(() => /** @type {SnippetFn} */ (snippet)(anchor, ...args));
+		} else if (!is_chain) {
+			e.invalid_snippet();
```

Comment:
Is there any drawback to just remove the if block instead? In my local testing this fails with a somewhat readable error message
```suggestion
		snippet_effect = branch(() => /** @type {SnippetFn} */ (snippet)(anchor, ...args));
```
Or at least, only have the more readable error message and extra argument at DEV time to save a few bytes? (possibly this means having a separate `snippet_dev` function)

Reply:
Then it won't error in prod? Do we want to have that difference?

Reply:
Why wouldn't it error in prod? The `if` block is the reason for the error, isn't it? And if we remove that, it will fail invoking the snippet inside the branch effect.

Reply:
The extra argument `is_chain` needs to be there for the error to work in dev or prod.

Reply:
Look at my code suggestion: It removes the if block altogether and always executes the `snippet_effect = branch(() => (snippet)(anchor, ...args))` line. This will then fail as soon as the branch is executed. Why does that not suffice?

Reply:
I did that originally but the error message is garbage. It says “snippet” is not a function in Chrome and even worse in other browsers.

Reply:
You're right it's not very helpful. I would lean towards showing the uglier error in PROD in order to save a few bytes - but if others feel differently (maybe we wait on a third opinion) I'm ok with keeping it like it's now.

Reply:
Personally, I would prefer a nicer message in DEV **and** fewer bytes in PROD. Maybe we could get the best of both worlds with a helper like this:
```js
function maybe_error() {
  if (Math.random() > 0.5) throw new Error() 
}

function explain_error(fn, error) {
  if (DEV) {
    try {
      fn();
    } catch (original) {
      error.cause = original;
      throw error;
    }
  } else {
    fn();
  }
}

explain_error(maybe_error, new Error("nice error"));
```

I believe this would optimize nicely in PROD.

Reply:
It's not the most helpful error but we don't do anything special for `undefined()` in normal code either. I think it's perfectly fine. I'd rather we have a dev-time error that includes the location of the offending tag than a less-useful error that also appears in prod

Reply:
I took an alternative approach and just remove the error altogether and instead made the compiler insert an empty snippet for when a nullish chain expression is used dynamically.